src/creeps/Zombie.cpp

```diff
@@ -1,61 +1,308 @@
+        }
 
-    return distr(eng);
+        // Changed to attack state once attack code is ready
+        if (state != ZombieState::ZOMBIE_IDLE) {
+            setState(ZombieState::ZOMBIE_IDLE);
+        }
+
+        return;
+    }
+
+    // Each case will check if the zombie is within bounds before moving
+    switch(d) {
+        case ZombieDirection::DIR_R:
+            if (checkBounds(startX + ZOMBIE_VELOCITY, startY)) {
+                setDX(ZOMBIE_VELOCITY);
+                setDY(0);
+                setAngle(static_cast<double>(ZombieAngles::EAST));
+            }
+            break;
+        case ZombieDirection::DIR_RD:
+            if (checkBounds(startX + ZOMBIE_VELOCITY, startY + ZOMBIE_VELOCITY)) {
+                setDX(ZOMBIE_VELOCITY);
+                setDY(ZOMBIE_VELOCITY);
+                setAngle(static_cast<double>(ZombieAngles::SOUTHEAST));
+            }
+            break;
+        case ZombieDirection::DIR_D:
+            if (checkBounds(startX, startY + ZOMBIE_VELOCITY)) {
+                setDX(0);
+                setDY(ZOMBIE_VELOCITY);
+                setAngle(static_cast<double>(ZombieAngles::SOUTH));
+            }
+            break;
+        case ZombieDirection::DIR_LD:
+            if (checkBounds(startX - ZOMBIE_VELOCITY, startY + ZOMBIE_VELOCITY)) {
+                setDX(-ZOMBIE_VELOCITY);
+                setDY(ZOMBIE_VELOCITY);
+                setAngle(static_cast<double>(ZombieAngles::SOUTHWEST));
+            }
+            break;
+        case ZombieDirection::DIR_L:
+            if (checkBounds(startX - ZOMBIE_VELOCITY, startY)) {
+                setDX(-ZOMBIE_VELOCITY);
+                setDY(0);
+                setAngle(static_cast<double>(ZombieAngles::WEST));
+            }
+            break;
+        case ZombieDirection::DIR_LU:
+            if (checkBounds(startX - ZOMBIE_VELOCITY, startY - ZOMBIE_VELOCITY)) {
+                setDX(-ZOMBIE_VELOCITY);
+                setDY(-ZOMBIE_VELOCITY);
+                setAngle(static_cast<double>(ZombieAngles::NORTHWEST));
+            }
+            break;
+        case ZombieDirection::DIR_U:
+            if (checkBounds(startX, startY - ZOMBIE_VELOCITY)) {
+                setDX(0);
+                setDY(-ZOMBIE_VELOCITY);
+                setAngle(static_cast<double>(ZombieAngles::NORTH));
+            }
+            break;
+        case ZombieDirection::DIR_RU:
+            if (checkBounds(startX + ZOMBIE_VELOCITY, startY - ZOMBIE_VELOCITY)) {
+                setDX(ZOMBIE_VELOCITY);
+                setDY(-ZOMBIE_VELOCITY);
+                setAngle(static_cast<double>(ZombieAngles::NORTHEAST));
+            }
+            break;
+        case ZombieDirection::DIR_INVALID:  // Shouldn't ever happens, gets rid of warning
+            break;
+    }
+
+    if (frame > 0) {
+        --frame;
+    } else {
+        setCurFrame(ZOMBIE_FRAMES);
+        ++step;
+    }
+
+    setCurDir(d);
+    if (state != ZombieState::ZOMBIE_MOVE) {
```

Comment:
If statement is not required
If the zombie isn't in a move state, it is now in a move state
If the zombie is in a move state, it is now in a move state
So we can remove a branch and always set the zombies state to moving